### PR TITLE
Create a Nested version of the css-vars mixin

### PR DESCRIPTION
**Pull request by RedLucas**
The idea of this is to allow you to apply variables in a nested context. So you can apply variables to a class, and not global. I couldn't think of a way to detect if we are root already so I made a second function and tried to take out the stuff from both and put it into its own function. 

Let me know what you think, but I think this is pretty cool.

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
-  "name": "css-var",
-  "version": "2.1.1",
+  "name": "css-vars",
```
Comment (shospodarets) [inline bower.json:2]: The name is already taken in bower, so it was not a typo, it was on purpose

File: css-vars.scss [hunk 2]
```diff
@@ -80,19 +80,8 @@ $css-vars-debug-log: false !default;
...
-// E.G.:
-// @include css-vars((
-//    --color: rebeccapurple,
-//    --height: 68px,
-//    --margin-top: calc(2vh + 20px)
-// ));
-///
-@mixin css-vars($varMap: null) {
-  // CHECK PARAMS
+@mixin _css-vars($varMap: null) {
```
Comment (shospodarets) [inline css-vars.scss:83]: In this case better to exclude the CHECK_PARAMS part and call the new mixin differently to reflect it

File: css-vars.scss [hunk 3]
```diff
@@ -120,4 +124,32 @@ $css-vars-debug-log: false !default;
...
+///
+// CSS mixin to provide variables in a nested context.
+// E.G.:
+// @include css-vars-nested((
+//    --color: rebeccapurple,
+//    --height: 68px,
+//    --margin-top: calc(2vh + 20px)
+// ));
+///
+@mixin css-vars-nested($varMap: null) {
```
Comment (shospodarets) [inline css-vars.scss:140]: The mixins css-vars and and css-vars-nested are the same,
except the parts "@at-root :root {" and "& {" - please consider extending that to follow the DRY

Comment (shospodarets) [reply css-vars.scss:140]: Can you please update the mixins comments to reflect the difference between them

Comment (shospodarets) [reply css-vars.scss:140]: And the last one- with such an update README should be changed with the provided addition

Comment (shospodarets) [inline css-vars.scss:140]: Can you consider adding tests for this as well

**Comment by shospodarets** [comment]
Hey @RedLucas 
Thank you for the PR.
I provided some comments.
Thank you for the idea, in general it's nice, just breaks the idea of having all the variables globally, to avoid problems.
Take a look into the postcss-css-variables discussions, why they store everything on the :root scope:
https://github.com/postcss/postcss-custom-properties/issues/1
https://github.com/postcss/postcss-custom-properties/issues/9

That's why I'm not sure this addition will not provide additional unexpected behavior.

File: bower.json (hunk 1 above)
Comment (RedLucas) [reply bower.json:2]: Ah, sorry, didn't know that.

**Comment by RedLucas** [comment]
I also get the concern regarding unexpected behavior, though I am using this and so far I've been happy with it. 

I need it to work like this for a project I'm working on, though I could always just use my fork... I'd prefer to contribute it back into this project if that's possible. 

Just need to keep using this thing until Microsoft drops support for IE11, then maybe we can ascend to a greater state of being.

**Comment by shospodarets** [comment]
Closing as the PR is not mergable and not updated for a year.

**Pull request closed without merging**